hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestReconstructStripedBlocks.java

```diff
@@ -515,4 +517,63 @@ public void testReconstrutionWithBusyBlock1() throws Exception {
     assertEquals(9, bm.countNodes(blockInfo).liveReplicas());
   }
 
+  @Test
+  public void testReconstructionWithStorageTypeNotEnough() throws Exception {
+    final HdfsConfiguration conf = new HdfsConfiguration();
+    conf.setInt(DFSConfigKeys.DFS_NAMENODE_REDUNDANCY_INTERVAL_SECONDS_KEY, 1);
+
+    // nine disk node eleven archive node
```

Comment:
Please use a capital letter at the beginning of the sentences and period at the end of annotation. (L525,L533,L556,L563)